backend/rollback.go

```diff
@@ -14,35 +13,27 @@ import (
 // RollbackApplication rollback application to previous version.
 func (b *Backend) RollbackApplication(applicationId string) error {
 	logrus.Infof("Rollback application %s", applicationId)
-	app, err := b.store.FetchApplication(applicationId)
+	app, err := b.store.GetApp(applicationId)
 	if err != nil {
 		return err
 	}
 
-	if app == nil {
-		logrus.Errorf("Application %s not found for rollback", applicationId)
-		return errors.New("Application not found")
-	}
-
 	// Update application status to ROLLINGBACK
-	if err := b.store.UpdateApplicationStatus(app.ID, "ROLLINGBACK"); err != nil {
+	if err := b.store.UpdateAppStatus(app.ID, "ROLLINGBACK"); err != nil {
 		return err
 	}
 
-	versions, err := b.store.ListApplicationVersions(applicationId)
+	versions, err := b.store.GetAndSortVersions(applicationId)
 	if err != nil {
 		return err
 	}
 
-	sort.Strings(versions)
-
-	rollbackVer := versions[len(versions)-2]
-	version, err := b.store.FetchApplicationVersion(applicationId, rollbackVer)
-	if err != nil {
-		return err
+	if len(versions) < 2 {
+		return errors.New("not have history version to rollback")
 	}
+	version := versions[len(versions)-2]
```

Comment:
How about define a struct represent group of versions?
Have a methods VersionGroup.CurrentVersion VersionGroup.LastVersion VersionGroup.LastHealthVersion